src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -860,6 +966,151 @@ std::shared_ptr<BaseSegment> ExpressionEvaluator::evaluate_expression_to_segment
   return segment;
 }
 
+PosList ExpressionEvaluator::evaluate_expression_to_pos_list(const AbstractExpression& expression) {
+  /**
+   * Only Expressions returning a Bool can be evaluated to a PosList of matches.
+   *
+   * In, Like and NotLike Expressions are evaluated by generating a Series of booleans
+   * (evaluate_expression_to_result<>()) which is then scanned for positive entries.
+   * TODO(anybody) Add fast implementations for In, Like and NotLike as well.
+   *
+   * All other Expression types have dedicated, hopefully fast implementations.
+   */
+
+  auto result_pos_list = PosList{};
+
+  switch (expression.type) {
+    case ExpressionType::Predicate: {
+      const auto& predicate_expression = static_cast<const AbstractPredicateExpression&>(expression);
+
+      switch (predicate_expression.predicate_condition) {
```

Comment:
Here is probably not the best place, but maybe in the TableScanImpl?

Reply:
One step further up, in the TableScan::get_impl(), this is documented nicely. And this is where I would expect this doc since the code path is chosen there.

Reply:
Documented, yes, but not checked. However, if we have tests that make sure that the impl is chosen correctly, that would probably be sufficient.

Reply:
We do have tests for this in the TableScan tests. How would you expect this to be checked? TableScan::get_impl() investigates the expression and if no Impl can handle it, it uses the ExpressionEvaluatorImpl.

Reply:
If this is already sufficiently tested, let's leave it the way it is.